tests/unit/test_project_structure.py

```diff
@@ -955,6 +955,43 @@ def test_delete_from_table(self):
 
         self.run_sql('delete from tbl1 where a=1', database='dummy_data')
 
+    def test_partition(self):
+        df = pd.DataFrame([
+            {'a': 1, 'b': dt.datetime(2020, 1, 1)},
+            {'a': 2, 'b': dt.datetime(2020, 1, 2)},
+            {'a': 3, 'b': dt.datetime(2020, 1, 3)},
+            {'a': 4, 'b': dt.datetime(2020, 1, 5)},
+            {'a': 5, 'b': dt.datetime(2020, 1, 6)},
+            {'a': 6, 'b': dt.datetime(2020, 1, 7)},
+        ])
+        self.save_file('tasks', df)
+
+        # -- create model --
+        self.run_sql(
+            '''
+                CREATE model mindsdb.task_model
+                from files (select * from tasks)
+                PREDICT a
+                using engine='dummy_ml',
+                join_learn_process=true
+            '''
+        )
+
+        # use model
+        ret = self.run_sql('''
+             SELECT *
+               FROM files.tasks as t
+               JOIN mindsdb.task_model as m
+               using partition_size=2
+        ''')
+
+        # the same rows in output
```

Comment:
Shouldn't we also check for the index and/or row order? Or is merge order not guaranteed?

Reply:
Yes, the order might be different.

Reply:
yes. planner don't split TS queries into partition at the moment